install.sh

```diff
@@ -88,6 +85,96 @@ execute_installation() {
         setup_shared_hook_repositories
         echo # For visual separation
     fi
+
+    thank_you
+}
+
+############################################################
+# Apply legacy fixes if appropriate.
+#
+# Returns:  0
+############################################################
+apply_legacy_fixes() {
+
+    # if not installed this will be empty
+    INSTALLED_VERSION=$(git hooks version 2>/dev/null | grep "Version:" | sed -E "s/^Version: //")
+
+    # Updating from this version needs a single install fixture
+    LEGACY_FIXTURE_SINGLE_INSTALL=$(echo "$INSTALLED_VERSION 2004.272234-8af36c" | awk "{ print (\$1 <= \$2) }")
+
+    if [ "$LEGACY_FIXTURE_SINGLE_INSTALL" = "1" ] &&
+        LEGACY_FIXTURE_SINGLE_INSTALL="true"; then
+        apply_legacy_fix_single_install
+    fi
+
+    return 0
+}
+
+############################################################
+# Apply single install fixes for deperecated single installs
+#
+# Returns:  0
+############################################################
+apply_legacy_fix_single_install() {
+    if [ -f "$INSTALL_DIR/autoupdate/registered" ]; then
+
+        echo "Apply single install fixture to non-single installs ... "
+
+        LIST="$INSTALL_DIR/autoupdate/registered"
+        IFS="$IFS_NEWLINE"
+        while read -r INSTALLED_REPO; do
+            unset IFS
+
+            if ! is_git_repo "$INSTALLED_REPO"; then
+                # Not existing git dir -> skip.
+                true
+
+            else
+                echo "Migrate githooks install in $INSTALLED_REPO"
+                install_hooks_into_repo "$INSTALLED_REPO"
+            fi
+
+            IFS="$IFS_NEWLINE"
+        done <"$LIST"
+
+        # Delete the old list.
+        rm -rf "$LIST"
+
+        WHAT="single"
+    else
+        WHAT="single & non-single"
+    fi
+
+    echo "!! MIGRATION INFORMATION: From now on single installs are deprecated: "
+    echo "!!"
+    echo "!! All still existing $WHAT install repositories:"
+    echo "!!   - \`install.sh --single\` or with "
+    echo "!!   - \`git hooks install\`"
+    echo "!! still have old githooks installed to them."
+    echo "!!"
+    echo "!! Because we cannot locate these installs, "
+    echo "!! you should use the \"install into existing repositories\""
+    echo "!! later on during this install procedure to reinstall into found"
+    echo "!! single install repositories for a clean up."
```

Comment:
This might be a bit confusing wording. If we cannot locate them, then how can we later fix them anyway? I think we *can* locatie them (by searching), but have no predefined list. So maybe "Because these installs were not kept in a list," and maybe "to search the filesytem for these repositories for a clean up"?

Also, I'm a bit confused about the meaning of `$WHAT` here. If there is no `autoupdate/registered` list, what does that mean? Isn't it just that non of the installed repos were marked for autoupdate? So why should this message change?

Finally: There *might* not be any leftover repositories, right? If all repos had autoupdate enabled, they would be fixed now. So maybe the message should clarify that?

Reply:
Also: the install into existing repositories might not actually run later (e.g. when run with `--single`, or `DO_UPDATE_ONLY`)?

Reply:
`autoupdate/registered` list unfortunately only contained **non-single** installs -> thats why we can do the most appropriate fix.
Good point: Maybe I should implement a search similar to the  `install into existing repos` right here
In both cases the user needs a search path to enter to search all repos to make **no leftover**

Reply:
Might be better to just tell the user a command they can run to search later?

Reply:
true